### PR TITLE
Increase retry for zipdeploy

### PR DESCRIPTION
**Pull request by weikanglim**
- Add additional ~12s retry delay by adding an additional retry attempt for zipdeploy

Helps with #1032

### CODE REVIEW

File: cli/azd/pkg/azsdk/zip_deploy_client.go [hunk 1]
```diff
@@ -65,6 +65,12 @@ func NewZipDeployClient(
 	// We do not have a Resource provider to register
 	options.DisableRPRegistration = true
 
+	// Increase default retry attempts from 3 to 4.
```
Comment (weikanglim) [inline cli/azd/pkg/azsdk/zip_deploy_client.go:68]:
```suggestion
	// Increase default retry attempts from 3 to 4 as zipdeploy often fails with 3 retries.
```

File: cli/azd/main.go [hunk 2]
```diff
@@ -39,7 +40,11 @@ func main() {
 
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 
-	if !isDebugEnabled() {
+	if isDebugEnabled() {
```
Comment (vhvb1989) [inline cli/azd/main.go:43]: ?  should this go in?

File: cli/azd/pkg/azsdk/zip_deploy_client.go [hunk 3]
```diff
@@ -65,6 +65,12 @@ func NewZipDeployClient(
 	// We do not have a Resource provider to register
 	options.DisableRPRegistration = true
 
+	// Increase default retry attempts from 3 to 4 as zipdeploy often fails with 3 retries.
+	// With the default azcore.policy options, this introduces up to 20 seconds of exponential back-off.
+	options.Retry = policy.RetryOptions{
+		MaxRetries: 4,
+	}
```
Comment (vhvb1989) [inline cli/azd/pkg/azsdk/zip_deploy_client.go:72]: We could also try adding a longer wait on the first error?  But 4 times is also good

**Review by vhvb1989 — APPROVED** [review]
the log listener should be in another PR

File: cli/azd/main.go (hunk 2 above)
Comment (weikanglim) [reply cli/azd/main.go:43]: Yes. Without it, we have no debugging story for any Azure requests currently.

File: cli/azd/pkg/azsdk/zip_deploy_client.go (hunk 3 above)
Comment (weikanglim) [reply cli/azd/pkg/azsdk/zip_deploy_client.go:72]: It's always a tradeoff. If we delay the early initial retry, we end up being slower on requests that might succeed due to a temporary flake.

File: cli/azd/main.go (hunk 2 above)
Comment (vhvb1989) [reply cli/azd/main.go:43]: We should have a different PR for that. With a title like `set log handler for azure sdk client options`

Comment (weikanglim) [reply cli/azd/main.go:43]: Will split

File: cli/azd/main.go [hunk 4]
```diff
@@ -39,7 +40,11 @@ func main() {
...
+		azcorelog.SetListener(func(event azcorelog.Event, msg string) {
+			log.Printf("%s: %s\n", event, msg)
```
Comment (vhvb1989) [inline cli/azd/main.go:45]: I would prefer not to have Azure SDK logs added by default to the azd logs.
IMHO, it should be an opt in configuration.  Something like `azd config logs azuresdkclient ON`

**Pull request merged**